### PR TITLE
Feature/get eks nodegroups lambda

### PR DESCRIPTION
**Pull request by stthoom**
### Description:

EKS Cluster node groups JL UI and backend lambda based SDK calls.

### Issue Reference URL

https://github.com/awslabs/aws-eks-data-maker/issues/<NUMBER>

----------
### Do not change content below. Mark applicable check boxes only.

Thank you for submitting a contribution to AWS Orbit Workbench.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a ISSUE associated with this PR?
- [ ] Does your PR title start with ISSUE-XXXX where XXXX is the issue number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [ ] Has your PR been rebased against the latest commit within the target branch (typically 'main')?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under Apache License 2.0?

### CODE REVIEW

File: cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    )
+                    if ltr["LaunchTemplateVersions"]:
+                        launch_template = ltr["LaunchTemplateVersions"][0]
+                        nodegroup_dict["launch_template_data"] = launch_template["LaunchTemplateData"][
+                            "BlockDeviceMappings"
+                        ]
+
+                return_response.append(nodegroup_dict)
+    except Exception as ekse:
+        logger.error("Error describing cluster %s nodegroups: %s", cluster_name, ekse)
```
Comment (abaror) [inline cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:64]: re-throw exception ?

File: cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+REGION: str = os.environ["REGION"]
+
+
+def get_nodegroups(cluster_name: str) -> List[Dict[str, Dict[str, str]]]:
+    return_response: List[Dict[str, Dict[str, str]]] = []
+    eks_client = boto3.client("eks")
+    ec2_client = boto3.client("ec2")
+    try:
+        nodegroups_response = eks_client.list_nodegroups(clusterName=cluster_name)
+        for nodegroup_name in nodegroups_response["nodegroups"]:
```
Comment (abaror) [inline cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:34]: should we do so much in the lambda ?  or just return almost what the list* gives you and do all that inside the client code?

File: sdk/aws_orbit_sdk/controller.py [hunk 3]
```diff
@@ -464,6 +464,29 @@ def list_storage_class():
...
+    env_name = props["AWS_ORBIT_ENV"]
+    try:
+        cluster_name = f"orbit-{env_name}"
+        _logger.debug(f"Fetching cluster {cluster_name} nodegroups")
+        response: Dict[str, Any] = boto3.client("lambda").invoke(
+            FunctionName=f"orbit-{env_name}-eks-service-handler",
+            InvocationType="RequestResponse",
+            Payload=json.dumps({"cluster_name": cluster_name}).encode("utf-8"),
+        )
+        _logger.debug(response)
```
Comment (abaror) [inline sdk/aws_orbit_sdk/controller.py:479]: remove these debug calls.. going to be hard to see anything if we have so much logging

File: cli/aws_orbit/remote_files/cdk/env.py [hunk 4]
```diff
@@ -385,6 +386,29 @@ def _create_token_validation_lambda(self) -> aws_lambda.Function:
...
+            runtime=aws_lambda.Runtime.PYTHON_3_8,
+            timeout=Duration.seconds(5),
+            environment={
+                "REGION": self.context.region,
+            },
+            initial_policy=[
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=["eks:List*", "eks:Describe*", "ec2:Describe*"],
+                    resources=["*"],
```
Comment (chamcca) [inline cli/aws_orbit/remote_files/cdk/env.py:407]: is it possible to scope this to just `orbit-*` resources?

File: cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py (hunk 2 above)
Comment (chamcca) [reply cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:34]: agreed. just returning the sdk response will reduce the amount of rework required when the vpc endpoint becomes available

Comment (stthoom) [reply cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:34]: Agreed. If we move list_nodegroups to client code, need to trigger lambda twice. 1) list_nodegroups 2) describe_nodegroup. 

ec2..describe_launch_template_versions can be moved to client code to use VPC endpoint.

Comment (abaror) [reply cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:34]: ok, I get it , I guess its not worth splitting into multiple calls, but whatever else can be reduced would be good.

Comment (stthoom) [reply cli/aws_orbit/remote_files/cdk/lambda_sources/eks_service_handler/index.py:33]: Moved ec2 calls to SDK. Lambda holds eks calls.

**Review by abaror — APPROVED** [review]

**Pull request merged**